pkg/kubelet/watchdog/watchdog_linux.go

```diff
@@ -0,0 +1,158 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchdog
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/coreos/go-systemd/v22/daemon"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/healthz"
+	"k8s.io/klog/v2"
+)
+
+// WatchdogClient defines the interface for interacting with the systemd watchdog.
+type WatchdogClient interface {
+	SdWatchdogEnabled(unsetEnvironment bool) (time.Duration, error)
+	SdNotify(unsetEnvironment bool) (bool, error)
+}
+
+// DefaultWatchdogClient implements the WatchdogClient interface using the actual systemd daemon functions.
+type DefaultWatchdogClient struct{}
+
+var _ WatchdogClient = &DefaultWatchdogClient{}
+
+func (d *DefaultWatchdogClient) SdWatchdogEnabled(unsetEnvironment bool) (time.Duration, error) {
+	return daemon.SdWatchdogEnabled(unsetEnvironment)
+}
+
+func (d *DefaultWatchdogClient) SdNotify(unsetEnvironment bool) (bool, error) {
+	return daemon.SdNotify(unsetEnvironment, daemon.SdNotifyWatchdog)
+}
+
+// Option defines optional parameters for initializing the healthChecker
+// structure.
+type Option func(*healthChecker)
+
+func WithWatchdogClient(watchdog WatchdogClient) Option {
+	return func(hc *healthChecker) {
+		hc.watchdog = watchdog
+	}
+}
+
+type healthChecker struct {
+	checkers     []healthz.HealthChecker
+	retryBackoff wait.Backoff
+	interval     time.Duration
+	watchdog     WatchdogClient
+}
+
+var _ HealthChecker = &healthChecker{}
+
+const notifyInterval = time.Nanosecond
```

Comment:
this is too small. Let's make our own minimal value. We can adjust it later if needed. But I want to protect kubelet from running too hot because of misconfiguration

```suggestion
const minimalNotifyInterval = time.Second
```

Reply:
ok